geo/geomfn: implement ST_Area, ST_Length, ST_Distance, ST_Perimeter

otan opened the pull request:
Release note (sql change): Added newly implemented functions `ST_Area`,
`ST_Length`, `ST_Distance` and `ST_Perimeter` for Geometry functions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48074)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 14 of 14 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48074#-:-M60vrRx1M4oaVJ7ufmy:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geos/geos_unix.cc, line 275 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48074#-M60uEBZASTPLjdlykSu:-M60uEBZASTPLjdlykSv:bivita0) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2ebd06756ae7414c51bd413c72cd36670dcc95d/pkg/geo/geos/geos_unix.cc#L275)):*
> ```cpp
>   *clippedEWKB = {.data = NULL, .len = 0};
> 
>   auto wkbReader = lib->GEOSWKBReader_create_r(handle);
> ```

`wkbReader` is unused.
___
*[pkg/geo/geos/geos_unix.cc, line 305 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48074#-M60v5ZvAL02hk3WTwIb:-M60v5ZvAL02hk3WTwIc:bbpgfat) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2ebd06756ae7414c51bd413c72cd36670dcc95d/pkg/geo/geos/geos_unix.cc#L305)):*
> ```cpp
>     if (r == 0) {
>       if (error.length() == 0) {
>         error.assign(CR_GEOS_NO_ERROR_DEFINED_MESSAGE);
> ```

do you know where things are going wrong in the GEOS code for it drop the error?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48074)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/geos/geos_unix.cc, line 275 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48074#-M60uEBZASTPLjdlykSu:-M61Npxq1yWlaZ1FRn8w:b-yomejz) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2ebd06756ae7414c51bd413c72cd36670dcc95d/pkg/geo/geos/geos_unix.cc#L275)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

`wkbReader` is unused.
</blockquote></details>

nice catch.
___
*[pkg/geo/geos/geos_unix.cc, line 305 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48074#-M60v5ZvAL02hk3WTwIb:-M61NrM59bmaTpIU1zvi:b-ae5p8l) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2ebd06756ae7414c51bd413c72cd36670dcc95d/pkg/geo/geos/geos_unix.cc#L305)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

do you know where things are going wrong in the GEOS code for it drop the error?
</blockquote></details>

it's always wrapped around a try catch, so if the error is dropped it's basically a code bug.


<!-- Sent from Reviewable.io -->

otan committed:
geo/geomfn: implement ST_Area, ST_Length, ST_Distance, ST_Perimeter

Release note (sql change): Added newly implemented functions `ST_Area`,
`ST_Length`, `ST_Distance` and `ST_Perimeter` for Geometry functions.

otan commented:
bors r=sumeerbhola
